src/main/java/com/atlassian/bitbucket/jenkins/internal/applink/oauth/serviceprovider/consumer/PersistentServiceProviderConsumerStore.java

```diff
@@ -240,8 +240,8 @@ public void marshal(Object source, HierarchicalStreamWriter writer, MarshallingC
             Consumer consumer = (Consumer) source;
             addNode(writer, CONSUMER_KEY, consumer.getKey());
             addNode(writer, CONSUMER_NAME, consumer.getName());
-            addNode(writer, CONSUMER_DESCRIPTION, consumer.getDescription());
-            addNode(writer, CALLBACK_URL, consumer.getCallback());
+            consumer.getDescription().ifPresent(description -> addNode(writer, CONSUMER_DESCRIPTION, description));
+            consumer.getCallback().ifPresent(callback -> addNode(writer, CALLBACK_URL, callback));
```

Comment:
I assume it is fine that the nodes are not present?

Reply:
Yup, I'm pretty sure (and we do test this) absent nodes are treated as null (in fact omission is the only way we can handle `null`, otherwise it puts a nice `null` string in the XML and then tries to unmarshall it as a string value of `null` and fails miserably; and likewise for `Optional.empty()`).